tests/integration_tests/conftest.py

```diff
@@ -121,6 +149,8 @@ def _client(request, fixture_utils, session_cloud):
         user_data=user_data, launch_kwargs=launch_kwargs
     ) as instance:
         yield instance
+        if integration_settings.COLLECT_LOGS:
```

Comment:
Since this is a fairly expensive operation, do we want to really do this on every instance, or gate it behind either a test failure or a force value?   This request kind of asks whether the COLLECT_LOGS value should be an enum instead of bool. values could by ALWAYS|ON_ERROR|NEVER?

what do folks think?

Reply:
I think the thinking was that for the SRU logs we'd need to collect the logs from most/all of the tests anyway. I could default it to False too. But having an ON_ERROR option seems like a good idea too. I'll take a look at adding that.

Reply:
Thanks a lot for this @TheRealFalcon +1.